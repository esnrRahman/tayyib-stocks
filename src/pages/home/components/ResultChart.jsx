import React from 'react';
import PropTypes from 'prop-types';
import { ResponsiveBar } from '@nivo/bar';

import StyledChartHolder from './styles/ResultChartStyles';

const ResultChart = ({ chartValues }) => (
  <StyledChartHolder>
    <ResponsiveBar
      data={
        [
          {
            'y-axis': 'Total Debt',
            pass: chartValues.debtPass,
            fail: chartValues.debtFail,
          },
          {
            'y-axis': 'Cash and equivalents',
            pass: chartValues.cashPass,
            fail: chartValues.cashFail,
          },
          {
            'y-axis': 'Accounts Receivable',
            pass: chartValues.receivablesPass,
            fail: chartValues.receivablesFail,
          },
        ]
      }
      keys={[
        'pass',
        'fail',
      ]}
      indexBy="y-axis"
      margin={{
        top: 0,
        right: 130,
        bottom: 50,
        left: 170,
      }}
      axisBottom={{
        tickSize: 5,
        tickPadding: 5,
        tickRotation: 0,
        legend: 'percentage (%)',
        legendPosition: 'middle',
        legendOffset: 40,
      }}
      padding={0.5}
      maxValue={100}
      colors="paired"
      colorBy="id"
      layout="horizontal"
      borderColor="inherit:darker(1.6)"
      labelTextColor="inherit:darker(1.6)"
      animate
      motionStiffness={90}
      motionDamping={15}
    />
  </StyledChartHolder>
);

ResultChart.propTypes = {
  chartValues: PropTypes.shape({
    debtPass: PropTypes.number.isRequired,
    debtFail: PropTypes.number.isRequired,
    cashPass: PropTypes.number.isRequired,
    cashFail: PropTypes.number.isRequired,
    receivablesPass: PropTypes.number.isRequired,
    receivablesFail: PropTypes.number.isRequired,
  }),
};

export default ResultChart;
