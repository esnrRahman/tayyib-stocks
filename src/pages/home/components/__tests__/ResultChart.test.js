import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ResultChart from '../ResultChart';

describe('<ResultChart />', () => {
  it('initial render', () => {
    const props = {
      chartValues: {
        debtPass: 10,
        debtFail: 0,
        cashPass: 33,
        cashFail: 10,
        receivablesPass: 15,
        receivablesFail: 0,
      },
    };
    const wrapper = shallow(
      <ResultChart
        {...props}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
