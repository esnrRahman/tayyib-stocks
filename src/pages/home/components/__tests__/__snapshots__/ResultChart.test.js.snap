// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultChart /> initial render 1`] = `
<styled.div>
  <ResponsiveBar
    animate={true}
    axisBottom={
      Object {
        "legend": "percentage (%)",
        "legendOffset": 40,
        "legendPosition": "middle",
        "tickPadding": 5,
        "tickRotation": 0,
        "tickSize": 5,
      }
    }
    borderColor="inherit:darker(1.6)"
    colorBy="id"
    colors="paired"
    data={
      Array [
        Object {
          "fail": 0,
          "pass": 10,
          "y-axis": "Total Debt",
        },
        Object {
          "fail": 10,
          "pass": 33,
          "y-axis": "Cash and equivalents",
        },
        Object {
          "fail": 0,
          "pass": 15,
          "y-axis": "Accounts Receivable",
        },
      ]
    }
    indexBy="y-axis"
    keys={
      Array [
        "pass",
        "fail",
      ]
    }
    labelTextColor="inherit:darker(1.6)"
    layout="horizontal"
    margin={
      Object {
        "bottom": 50,
        "left": 170,
        "right": 130,
        "top": 0,
      }
    }
    maxValue={100}
    motionDamping={15}
    motionStiffness={90}
    padding={0.5}
  />
</styled.div>
`;
